src/main/java/org/cyclonedx/CycloneDxSchema.java

```diff
@@ -153,22 +152,34 @@ public Schema getXmlSchema(InputStream... inputStreams) throws SAXException {
         return schemaFactory.newSchema(schemaFiles);
     }
 
-    private org.everit.json.schema.Schema getJsonSchema12(boolean strict) throws IOException, URISyntaxException {
+    private org.everit.json.schema.Schema getJsonSchema12(boolean strict) throws URISyntaxException {
         if (strict) {
-            return getJsonSchema(this.getClass().getClassLoader().getResourceAsStream("bom-1.2-strict.schema.json"));
+            return getJsonSchema(
+                this.getClass().getClassLoader().getResourceAsStream("bom-1.2-strict.schema.json"),
+                this.getClass().getClassLoader().getResourceAsStream("ext/vulnerability-1.0-types.schema.json"),
+                this.getClass().getClassLoader().getResourceAsStream("ext/vulnerability-1.0-strict.schema.json")
+            );
         } else {
-            return getJsonSchema(this.getClass().getClassLoader().getResourceAsStream("bom-1.2.schema.json"));
+            return getJsonSchema(
+                this.getClass().getClassLoader().getResourceAsStream("bom-1.2.schema.json"),
+                this.getClass().getClassLoader().getResourceAsStream("ext/vulnerability-1.0-types.schema.json"),
+                this.getClass().getClassLoader().getResourceAsStream("ext/vulnerability-1.0.schema.json")
+            );
         }
     }
 
-    private org.everit.json.schema.Schema getJsonSchema(InputStream inputStream) throws IOException, URISyntaxException {
-        final JSONObject rawSchema = new JSONObject(new JSONTokener(inputStream));
-        final InputStream spdxInstream = this.getClass().getClassLoader().getResourceAsStream("spdx.schema.json");
-        final JSONObject spdxSchema = new JSONObject(new JSONTokener(spdxInstream));
-        final SchemaLoader schemaLoader = SchemaLoader.builder()
-                .registerSchemaByURI(new URI("http://cyclonedx.org/schema/spdx.schema.json"), spdxSchema)
-                .schemaJson(rawSchema)
-                .build();
-        return schemaLoader.load().build();
+    private org.everit.json.schema.Schema getJsonSchema(InputStream bomSchemaStream, InputStream... extSchemaStreams) throws URISyntaxException {
+        final JSONObject spdxSchema = new JSONObject(new JSONTokener(
+            this.getClass().getClassLoader().getResourceAsStream("spdx.schema.json")));
+        final SchemaLoaderBuilder schemaLoaderBuilder = SchemaLoader.builder()
+            .registerSchemaByURI(new URI(spdxSchema.get("$id").toString()), spdxSchema);
+
+        for (int i=0; i<extSchemaStreams.length; i++) {
```

Comment:
minor: not sure how strict this repo adhere to formatting, but the formatting (spacing) here  doesn't look correct to me